models/demos/t3000/falcon40b/tt/falcon_model.py

```diff
@@ -157,31 +186,8 @@ def model_preprocessing(self, llm_mode, input_ids, kv_cache_len, num_input_token
             assert sequence_size % 32 == 0, "For prefill, sequence_size must be multiple of 32!"
             assert kv_cache_len == 0, "For prefill, no kv_cache is passed in!"
 
-            attention_mask_bool = torch.ones(batch_size, 1, sequence_size, sequence_size, dtype=bool)
-            attention_mask_bool = attention_mask_bool.triu(diagonal=1)
-
-            attention_mask_memconfig = self.model_config["ATTN_MASK_MEMCFG"]
-            if attention_mask_memconfig.is_sharded():
-                attn_mask_shard_shape = attention_mask_memconfig.shard_spec.shape
-                attn_mask_shard_shape[-1] = sequence_size
-                attention_mask_memconfig.shard_spec.shape = attn_mask_shard_shape
+            tt_attention_mask = self.slice_attn_mask(sequence_size)
```

Comment:
Can we extend this for decode too?

Reply:
Not the same as for prefill so would leave it for another PR since do not have enough time at the moment since I've already run all the tests.